core/src/heaviest_subtree_fork_choice.rs

```diff
@@ -498,6 +498,44 @@ impl HeaviestSubtreeForkChoice {
         );
     }
 
+    fn heaviest_slot_on_same_voted_fork(&self, tower: &Tower) -> Option<Slot> {
+        tower
+            .last_voted_slot()
+            .map(|last_voted_slot| {
+                let heaviest_slot_on_same_voted_fork = self.best_slot(last_voted_slot);
+                if heaviest_slot_on_same_voted_fork.is_none() {
+                    if !tower.is_stray_last_vote() {
+                        // Unless last vote is stray, we always must be able to fetch last_voted_slot's
+                        // bast_slot, justifying to panic! here.
```

Comment:
`last_slot`

Reply:
Well, this should be `best_slot`, I rewritten the comment: 0bfb8227e04077e10ae5107d47569fccefaba648